### PR TITLE
Add host config to the docker run config

### LINKED ISSUE #977: Clicking "Show advanced run configuration" in Prefect UI erases RunConfig.host_config
## Description

A flow with `run_config=DockerRun(host_config={'network_mode': 'host'})` will have it's host_config overwritten if 'Show advanced run configuration' is clicked in the Prefect UI > Flow > Run.

## Expected Behavior

1. Either host_config should show for Docker run configurations in Prefect UI > Flow > Run
or
2. host_config should not be erased.

## Reproduction

```
with Flow(
    'dummy-flow',
    run_config=DockerRun(
        image='busybox',
        host_config={
            'network_mode': 'host',
        },
    ),
```

1. Register this flow
2. Go to flow > Run
3. Click 'Show advanced run configuration', don't change anything.
4. Click run.
5. See that resulting DockerRun on DockerAgent does not honor network_mode host. (possible to inspect by hacking the agent code before starting, f.e. adding a few print statements to https://github.com/PrefectHQ/prefect/blob/master/src/prefect/agent/docker/agent.py#L412)

## Environment

```
{
  "config_overrides": {},
  "env_vars": [
    "PREFECT__SERVER__HOST",
    "PREFECT__BACKEND"
  ],
  "system_information": {
    "platform": "Linux-5.4.0-80-generic-x86_64-with-debian-bullseye-sid",
    "prefect_backend": "server",
    "prefect_version": "0.15.3",
    "python_version": "3.6.13"
  }
}
```

### PR DESCRIPTION
**Pull request by znicholasbrown**
PR Checklist:

- [ ] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Adds the `host_config` field to the static run config options.

Resolves #977

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 ### Bugfixes
 
-- None
+- Add hsot config to the docker run config - [#988](https://github.com/PrefectHQ/ui/pull/988)
```
Comment (zhen0) [inline Changelog.md:11]:
```suggestion
- Add host config to the docker run config - [#988](https://github.com/PrefectHQ/ui/pull/988)
```

File: src/utils/runConfigs.js [hunk 2]
```diff
@@ -55,6 +55,13 @@ export const DockerRun = {
       input_type: 'object',
       label: 'Environment variables',
       description: 'Additional environment variables to set in the container.'
+    },
+    {
+      arg: 'host_config',
+      input_type: 'object',
+      label: 'Host Config',
+      description:
+        'A dictionary or runtime arguments to pass to the Docker Agent. See link below for options.'
```
Comment (zhen0) [inline src/utils/runConfigs.js:64]: Where's the link?

**Review by zhen0 — CHANGES_REQUESTED** [review]
One question and one small change @znicholasbrown

File: Changelog.md (hunk 1 above)
Comment (znicholasbrown) [reply Changelog.md:11]: Oops, nice spot 😅

File: src/utils/runConfigs.js (hunk 2 above)
Comment (znicholasbrown) [reply src/utils/runConfigs.js]: Oops, didn't want to include a link there, updated 👍🏻

**Review by zhen0 — APPROVED** [review]
Thanks for the changes @znicholasbrown

**Pull request merged**